aws/resource_aws_launch_template.go

```diff
@@ -57,8 +57,16 @@ func resourceAwsLaunchTemplate() *schema.Resource {
 			},
 
 			"default_version": {
-				Type:     schema.TypeInt,
-				Computed: true,
+				Type:          schema.TypeInt,
+				Optional:      true,
+				Computed:      true,
+				ConflictsWith: []string{"update_default_version"},
+			},
+
+			"update_default_version": {
```

Comment:
The `CustomizeDiff` function needs two updates:

- Ignoring `update_default_version` in the `latest_version` attribute `Computed` handling
- Adding `default_version` attribute `Computed` handling, similar to the `latest_version` attribute `Computed` handling (minus the inclusion of `update_default_version`)

This triggers downstream updates for resources referencing these attributes, so practitioners do not need to run `terraform apply` twice.

Reply:
@bflad I ran into a test failure in `TestAccAWSLaunchTemplate_defaultVersion` (step 3/3) where we expect to go from `default_version = 1, latest_version = 2` to `default_version=2, latest_version=2` (and the launchtemplatedata remains unchanged) but with this `CustomizeDiff`, the update doesn't occur. 

Does it make sense to check that the `update_default_version` value was explicitly set instead? e.g.
```go
customdiff.ComputedIf("default_version", func(diff *schema.ResourceDiff, meta interface{}) bool {
				for _, changedKey := range diff.GetChangedKeysPrefix("") {
					switch changedKey {
					case "name", "name_prefix", "description":
						continue
					default:
						if v, ok := diff.GetOk("update_default_version"); ok {
							return v.(bool) || diff.HasChange("default_version")
						}
					}
				}
				return false
			}),
```

Reply:
I think these are the scenarios we need to check:

1. `update_default_version` is enabled and `default_version` is not equal to `latest_version`
1. `update_default_version` is enabled and any versioned attribute changes
1. `default_version` changes

The top scenario is not covered in either of our logic, so maybe something like the below to shortcut that outside the versioned attribute checking:

```go
			customdiff.ComputedIf("default_version", func(diff *schema.ResourceDiff, meta interface{}) bool {
				if diff.Get("update_default_version").(bool) && diff.Get("default_version").(int) != diff.Get("latest_version").(int) {
					return true
				}

				for _, changedKey := range diff.GetChangedKeysPrefix("") {
					switch changedKey {
					case "name", "name_prefix", "description":
						continue
					default:
						return diff.Get("update_default_version").(bool) || diff.HasChange("default_version")
					}
				}
				return false
			}),
```

Reply:
Note that in the above scenarios, its also a little more clear that we can actually attempt to tell Terraform the expected value for the `Computed` attribute so it fills things correctly in downstream in the plan output instead of showing unknown, but that is certainly out of scope for this PR. 😉

Reply:
i think I'm still a bit confused with how we handle `default_version` though 😅 because in the test I wrote, theres a case where the value is set to `"2"` in the config and with the proposed logic, at Update time, Terraform logs show the value is changing from `"1" -> "<computed>"` when in fact the config had this new value set to `"2"` and as a result an update doesn't occur. is this what we should expect? or maybe the test case is missing something?

Reply:
Maybe we should try removing the `diff.HasChanges("default_version")` handling from the `CustomizeDiff`

Reply:
i think how you originally suggested in your first comment makes sense now (w/o the case for `update_default_version`) if i'm understanding it correctly, we want to tell Terraform the expected value for a `Computed` field only if its not possible to determine from a provided config ?

so in the case we hit this part:
```
if diff.Get("update_default_version").(bool) && diff.Get("default_version").(int) != diff.Get("latest_version").(int) {
					return true
				}
```
we'll again run into the issue where maybe a user wants to set the `default_version` to a previous version.

Reply:
latest commit has changes to the `CustomizeDiff`